app/server.go

```diff
@@ -261,6 +261,21 @@ func NewServer(options ...Option) (*Server, error) {
 		s.StartElasticsearch()
 	}
 
+	s.AddConfigListener(func(oldConfig *model.Config, newConfig *model.Config) {
+		if *oldConfig.GuestAccountsSettings.Enable && !*newConfig.GuestAccountsSettings.Enable {
+			if appErr := s.FakeApp().DeactivateGuests(); appErr != nil {
```

Comment:
It would be nice not to use `FakeApp` in new code.

Reply:
Changed to an initialization of the app inside the config change handler.